subprojects/gradle/performance/src/main/kotlin/com/avito/performance/PerformancePlugin.kt

```diff
@@ -112,29 +115,58 @@ open class PerformancePlugin : Plugin<Project> {
                                 group = TASK_GROUP
-                                description = "Download performance report from last build"
-                                graphiteKey.set(targetBranch)
+                                description = "Download performance report from target branch build"
+                                graphiteKey.set(
+                                    gitState.map {
+                                        // fail fast for more meaningful error message
+                                        requireNotNull(it.targetBranch?.name)
+                                        { "Can't run performance tasks without targetBranch specified" }
+                                    })
                                 performanceTests.set(
                                     File(
                                         performanceOutputDir,
                                         "previous_${performanceConfig.name}_${performanceResultsFileName}"
                                     )
                                 )
-                                this.reportCoordinates.set(reportCoordinates.copy(jobSlug = "${reportCoordinates.jobSlug}-${InstrumentationTestsAction.RUN_ON_TARGET_BRANCH_SLUG}"))
                                 buildId.set(envArgs.build.id.toString())
                                 reportApiUrl.set(instrumentationConfig.reportApiUrl)
                                 reportApiFallbackUrl.set(instrumentationConfig.reportApiFallbackUrl)
-                                dependsOn(performanceCollectProvider)
+
+                                @Suppress("WHEN_ENUM_CAN_BE_NULL_IN_JAVA")
+                                when (extension.targetBranchResultSource.get()) {
+                                    RUN_IN_PROCESS -> {
+                                        //run on target branch is a part of the same task in `instrumentation plugin`
+                                        dependsOn(sourceBranchResultsCollector)
+
+                                        buildId.set(envArgs.build.id.toString())
+
+                                        this.reportCoordinates.set(
+                                            reportCoordinates.copy(
+                                                jobSlug = "${reportCoordinates.jobSlug}-$RUN_ON_TARGET_BRANCH_SLUG"
+                                            )
+                                        )
+                                    }
+                                    FETCH_FROM_OTHER_BUILD -> {
+
+                                        //there is no need to depend on tests, it is a hack to postpone fetching
+                                        dependsOn(sourceBranchResultsCollector)
+
+                                        //todo probably it's a mistake; but we need to fetch teamcity API otherwise
+                                        buildId.set(null as String?)
+
+                                        this.reportCoordinates.set(
```

Comment:
I didn't get how it works now.
We have develop build produces the target data
And have PR build that consumes the target data
So this logic means we will consume target data from `originalBranch.commit` `extension.targetBuildConfigId`
What are the `originalBranch.commit` and `extension.targetBuildConfigId`?
Who, where, what and when did set `extension.targetBuildConfigId`?

Reply:
it was a mistake, i needed `targetBranch.commit`, not a `originalBranch.commit`
also `targetBuildConfigId` intended to be set from extension in application's `build.gradle`